hcipy/aperture/generic.py

```diff
@@ -249,7 +253,7 @@ def make_obstruction(aperture):
 	'''
 	return lambda grid: 1 - aperture(grid)
 
-def make_segmented_aperture(segment_shape, segment_positions, segment_transmissions=1):
+def make_segmented_aperture(segment_shape, segment_positions, segment_transmissions=1, return_segments=False):
```

Comment:
One likely change is a different way of getting the segments. Right now, this is part of the make_segmented_aperture() function, but this doesn't really work, as that function is supposed to return a field generator only. Now, if `return_segments` is True, the "Field generator" returns both a Field and a ModeBasis, so technically it is not a Field generator anymore. It may be better to have a separate function, or maybe even better, have the function `make_segmented_aperture()` return both a Field generator and a ModeBasis generator, or something.